### PR TITLE
[frontend] fix unknown entities in investigationGraph (#issue/4236)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Proposed changes

* Add entities to the queries
* Handle the edit of entities details

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4236

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx [hunk 1]
```diff
@@ -342,7 +357,23 @@ const StixDomainObjectEditionContainer = (props) => {
...
             <Form style={{ margin: '20px 0 20px 0' }}>
-              {'name' in stixDomainObject && (
+              {'result_name' in stixDomainObject ? (
+                <Field
+                  component={TextField}
+                  variant="standard"
+                  name="result_name"
+                  label={t('Result Name')}
+                  fullWidth={true}
+                  disabled={typesWithoutName.includes(
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: not sure this test with `typesWithoutName` is necessary here

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx [hunk 2]
```diff
@@ -342,7 +357,23 @@ const StixDomainObjectEditionContainer = (props) => {
...
+                    stixDomainObject.entity_type,
+                  )}
+                  onFocus={handleChangeFocus}
+                  onSubmit={handleResultName}
+                  helperText={
+                    <SubscriptionFocus context={editContext} fieldName="result_name" />
+                  }
+                />
+              ) : (
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: don't we need a test here for 'name' like it was before ? or was is unnecessary ?

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx [hunk 3]
```diff
@@ -297,8 +309,11 @@ const StixDomainObjectEditionContainer = (props) => {
   let initialValues = R.pipe(
     R.assoc('createdBy', createdBy),
     R.assoc('objectMarking', objectMarking),
-    R.pick(['name', 'description', 'createdBy', 'objectMarking', 'confidence']),
+    R.pick(['name', 'result_name', 'description', 'createdBy', 'objectMarking', 'confidence']),
   )(stixDomainObject);
+  if ('result_name' in stixDomainObject && stixDomainObject.result_name !== undefined) {
+    initialValues = R.assoc('result_name', stixDomainObject.result_name, initialValues);
+  }
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: This is already handle by R.pick([... , 'result_name', ...]) no ?

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx (hunk 1 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: Indeed, this test should be enough

'result_name' in stixDomainObject

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx [hunk 4]
```diff
@@ -563,6 +594,10 @@ const StixDomainObjectEditionFragment = createFragmentContainer(
           description
           aliases
         }
+        ... on MalwareAnalysis {
+          result_name
+          product
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: Where are we using the product field ?

File: opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationGraph.jsx [hunk 5]
```diff
@@ -166,6 +166,9 @@ const investigationGraphStixCoreObjectQuery = graphql`
         first_seen
         last_seen
       }
+      ... on MalwareAnalysis {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationGraph.jsx:169]: Maybe, you can try to use the representative mechanism here (a bit challenging I suppose)

I see usage of defaultValue in Graph.buildGraphData method

File: opencti-platform/opencti-front/src/utils/graph/EntityDetails.tsx [hunk 6]
```diff
@@ -142,6 +142,15 @@ const entityDetailsQuery = graphql`
         name
         x_mitre_id
         description
+        killChainPhases {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/utils/graph/EntityDetails.tsx]: Maybe, you can try to use the representative mechanism here.
Less complicated that the previous point (I think)

I see usage of defaultValue

Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/utils/graph/EntityDetails.tsx]: Where killChanPhases is used ?

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx (hunk 2 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: You're right, I didn't reproduce the case where that test is necessary, but I think it's better to add it the way it was before !

**Comment by Archidoit** [comment]
I still have 'Unknown' for the 'object' field just after adding a malware analysis in an investigation graph. If I refresh the page, the 'unknown' disapears.
image

When adding a malware analysis in an investigation graph, the names are not displayed:
image

**Comment by Archidoit** [comment]
KillChainPhase value is also Unknown just after an Expand, but has a value after refreshing the page.
image

**Comment by SarahBocognano** [comment]
Update: Comments has been resolved for the functionnal parts. Evaluating the time that refacto with representative will take

**Comment by RomuDeuxfois** [comment]
> Update: Comments has been resolved for the functionnal parts. Evaluating the time that refacto with representative will take

There are some changes in this PR https://github.com/OpenCTI-Platform/opencti/pull/4289
We can talk about it if you want

@RomuDeuxfois It's seems that your changes with the change of representative to StixObject will resolve the unknown entities in investigation.

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx [hunk 7]
```diff
@@ -297,7 +309,7 @@ const StixDomainObjectEditionContainer = (props) => {
   let initialValues = R.pipe(
     R.assoc('createdBy', createdBy),
     R.assoc('objectMarking', objectMarking),
-    R.pick(['name', 'description', 'createdBy', 'objectMarking', 'confidence']),
+    R.pick(['name', 'result_name', 'description', 'product', 'createdBy', 'objectMarking', 'confidence']),
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: Need to remove product, it's not used

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx (hunk 4 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectEditionOverview.jsx]: No need

**Comment by SarahBocognano** [comment]
Update: After thoughts, this issue should be merge that way. The refacto for pain point can wait a little. Feel free to review and test !

File: opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationAddStixCoreObjectsLines.jsx [hunk 8]
```diff
@@ -230,6 +231,7 @@ class InvestigationAddStixCoreObjectsLinesInvestigation extends Component {
                               content={
                                 stixCoreObject.description
                                 || fd(stixCoreObject.created_at)
+                                || stixCoreObject.product
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationAddStixCoreObjectsLines.jsx]: No need ?

File: opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationAddStixCoreObjectsLines.jsx [hunk 9]
```diff
@@ -429,6 +431,10 @@ const InvestigationAddStixCoreObjectsLines = createPaginationContainer(
                 first_seen
                 last_seen
               }
+              ... on MalwareAnalysis {
+                result_name
+                product
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/workspaces/investigations/InvestigationAddStixCoreObjectsLines.jsx]: No need ?

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**